pkg/web/api.go

```diff
@@ -82,15 +84,158 @@ func (api *API) RegisterFilters(filters ...Filter) {
+	logrus.Debugf("Replacing filter %s with %s", replacedFilterName, filter.Name())
+	registeredFilterPosition := api.findFilterPositionOrDie(replacedFilterName)
+	api.Filters[registeredFilterPosition] = filter
+}
+
+func (api *API) findFilterPosition(filterName string) int {
+	for i := range api.Filters {
+		registeredFilter := api.Filters[i]
+		if registeredFilter.Name() == filterName {
+			return i
+		}
+	}
+	return -1
+}
+
+func (api *API) findFilterPositionOrDie(filterName string) int {
+	registeredFilterPosition := api.findFilterPosition(filterName)
+	if registeredFilterPosition < 0 {
+		logrus.Panicf("Filter with name %s is not registered", filterName)
+	}
+	return registeredFilterPosition
+}
+
 // RegisterPlugins registers a set of plugins
 func (api *API) RegisterPlugins(plugins ...Plugin) {
 	for _, plugin := range plugins {
-		pluginSegments := api.decomposePlugin(plugin)
-		if len(pluginSegments) == 0 {
-			logrus.Panicf("%T does not implement any plugin operation", plugin)
+		pluginSegments := api.decomposePluginOrDie(plugin)
+		api.RegisterFilters(pluginSegments...)
+	}
+}
+
+func (api *API) RegisterPluginBefore(name string, plugin Plugin) {
+	logrus.Debugf("Registering plugin %s before %s", plugin.Name(), name)
+	lowestFilterName := ""
+	lowestFilterPos := math.MaxInt64
+	for i := range api.Filters {
+		filter := api.Filters[i]
+		if strings.HasPrefix(filter.Name(), name+":") && i < lowestFilterPos {
+			lowestFilterPos = i
+			lowestFilterName = filter.Name()
+		}
+	}
+	api.registerPluginRelatively(lowestFilterName, plugin, api.RegisterFilterBefore)
+}
+
+func (api *API) RemoveFilter(name string) {
+	position := api.findFilterPosition(name)
+	if position >= 0 {
+		api.Filters = append(api.Filters[:position], api.Filters[position+1:]...)
+	}
+}
+
+func (api *API) RegisterPluginAfter(name string, plugin Plugin) {
+	logrus.Debugf("Registering plugin %s after %s", plugin.Name(), name)
+	highestFilterName := ""
+	highestFilterPos := math.MinInt64
+	for i := range api.Filters {
+		filter := api.Filters[i]
+		if strings.HasPrefix(filter.Name(), name+":") && i > highestFilterPos {
+			highestFilterPos = i
+			highestFilterName = filter.Name()
 		}
+	}
+	api.registerPluginRelatively(highestFilterName, plugin, api.RegisterFilterAfter)
+}
 
-		api.RegisterFilters(pluginSegments...)
+func (api *API) registerPluginRelatively(relativeName string, plugin Plugin, relationFunc func(string, Filter)) {
+	pluginSegments := api.decomposePluginOrDie(plugin)
+	for i := range pluginSegments {
+		pluginSegment := pluginSegments[i]
+		relationFunc(relativeName, pluginSegment)
+	}
+}
+
+func (api *API) decomposePluginOrDie(plugin Plugin) []Filter {
+	pluginSegments := api.decomposePlugin(plugin)
+	if len(pluginSegments) == 0 {
+		logrus.Panicf("%T does not implement any plugin operation", plugin)
+	}
+	return pluginSegments
+}
+
+func (api *API) ReplacePlugin(name string, plugin Plugin) {
```

Comment:
Replace only if after decomposing the plugin you get segments that are the same (same count and same type and same order) as the segments of the plugin that is being replaced